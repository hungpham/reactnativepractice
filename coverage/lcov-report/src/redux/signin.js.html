<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux/signin.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/redux</a> signin.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.05% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.05% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/19</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import AuthenticationService from 'network/AuthenticationService';
&nbsp;
//=============================//
//      Action Types
//=============================//
const REQUEST = 'Request';
const SUCCESS = 'Success';
const FAILED  = 'Failed';
&nbsp;
//=============================//
//      Action Creators
//=============================//
function <span class="fstat-no" title="function not covered" >loginRequest(</span>) {
<span class="cstat-no" title="statement not covered" >	return {</span>
		type: REQUEST
	};
}
&nbsp;
function <span class="fstat-no" title="function not covered" >loginRequestSuccess(</span>json) {
<span class="cstat-no" title="statement not covered" >	return {</span>
		type: SUCCESS,
		payload: json
	};
}
&nbsp;
function <span class="fstat-no" title="function not covered" >loginRequestFailed(</span>error) {
<span class="cstat-no" title="statement not covered" >	return {</span>
		type: FAILED,
		error: error.message
	};
}
&nbsp;
//=============================//
//      Reducer
//=============================//
export const INITIAL_STATE = {
	error: '',
	loading: false
};
&nbsp;
export function <span class="fstat-no" title="function not covered" >signInReducer(</span>state = <span class="branch-0 cbranch-no" title="branch not covered" >INITIAL_STATE,</span> action) {
<span class="cstat-no" title="statement not covered" >  switch (action.type) {</span>
    case REQUEST:
<span class="cstat-no" title="statement not covered" >			return {</span>
				...state,
				loading: true,
				error: ''
			};
		case SUCCESS:
<span class="cstat-no" title="statement not covered" >			return {</span>
				...state,
				loading: false,
				error: '',
				user: action.payload
			};
		case FAILED:
<span class="cstat-no" title="statement not covered" >			console.log(action.error);</span>
<span class="cstat-no" title="statement not covered" >			return {</span>
				...state,
				loading: false,
				error: action.error
			};
		default:
<span class="cstat-no" title="statement not covered" >      return state;</span>
  }
};
&nbsp;
//=============================//
//4: Load Data
//=============================//
export function <span class="fstat-no" title="function not covered" >login(</span>userCredentials) {
<span class="cstat-no" title="statement not covered" >  return <span class="fstat-no" title="function not covered" >(d</span>ispatch, getState) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    dispatch(loginRequest());</span>
<span class="cstat-no" title="statement not covered" >    return AuthenticationService.signin(userCredentials)</span>
    .then(<span class="fstat-no" title="function not covered" >js</span>on =&gt; {
<span class="cstat-no" title="statement not covered" >        dispatch(loginRequestSuccess(json));</span>
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>ror =&gt; {
<span class="cstat-no" title="statement not covered" >      console.log('There has been a problem with your fetch operation: ' + error.message);</span>
<span class="cstat-no" title="statement not covered" >      dispatch(loginRequestFailed(error))</span>
    });
  };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Wed Mar 01 2017 17:43:58 GMT+0700 (ICT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
