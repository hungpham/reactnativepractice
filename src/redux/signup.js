import AuthenticationService from 'network/AuthenticationService';

//=============================//
//      Action Types
//=============================//
export const REQUEST = 'SignUp/SignUp_Request';
export const SUCCESS = 'SignUp/SignUp_Success';
export const FAILED  = 'SignUp/SignUp_Failed';

//=============================//
//      Action Creators
//=============================//
export function signupRequest() {
	return {
		type: REQUEST
	};
}

export function signupRequestSuccess(json) {
	return {
		type: SUCCESS,
		payload: json
	};
}

export function signupRequestFailed(error) {
	return {
		type: FAILED,
		error: error.message
	};
}

//=============================//
//      Reducer
//=============================//
export const INITIAL_STATE = {
	error: '',
	loading: false
};

export function signUpReducer(state = INITIAL_STATE, action) {
	console.log('signUpReducer action', action)
  switch (action.type) {
    case REQUEST:
			return {
				...state,
				loading: true,
				error: ''
			};
		case SUCCESS:
			return {
				...state,
				loading: false,
				error: '',
				user: action.payload
			};
		case FAILED:
			return {
				...state,
				loading: false,
				error: action.error
			};
		default:
      return state;
  }
};

//=============================//
//4: Load Data
//=============================//
export function signup(userCredentials) {
  return (dispatch, getState) => {
    // dispatch(signupRequest());
    return AuthenticationService.signUp(userCredentials)
    .then(json => {
        dispatch(signupRequestSuccess(json));
    })
    .catch(error => {
      console.log('There has been a problem with your fetch operation: ' + error.message);
      dispatch(signupRequestFailed(error))
    });
  };
}
