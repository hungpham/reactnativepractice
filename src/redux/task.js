import Task from 'network/Task';

//=============================//
//      Action Types
//=============================//
export const REQUEST = 'TASK/Request';
export const SUCCESS = 'TASK/Success';
export const FAILED  = 'TASK/Failed';

//=============================//
//      Action Creators
//=============================//
export function taskRequest() {
	return {
		type: REQUEST
	};
}

export function taskRequestSuccess(json) {
	return {
		type: SUCCESS,
		payload: json
	};
}

export function taskRequestFailed(error) {
	return {
		type: FAILED,
		error: error.message
	};
}

//=============================//
//      Reducer
//=============================//
export const INITIAL_STATE = {
	error: '',
	loading: false
};

export function taskReducer(state = INITIAL_STATE, action) {
  switch (action.type) {
    case REQUEST:
			return {
				...state,
				loading: true,
				error: ''
			};
		case SUCCESS:
			return {
				...state,
				loading: false,
				error: '',
				tasks: action.payload
			};
		case FAILED:
			return {
				...state,
				loading: false,
				error: action.error
			};
		default:
      return state;
  }
};

//=============================//
//4: Load Data
//=============================//
export function getList(userCredentials) {
  return (dispatch, getState) => {
    dispatch(taskRequest());
    return Task.getTasks(userCredentials)
    .then(json => {
        dispatch(taskRequestSuccess(json));
    })
    .catch(error => {
      console.log('There has been a problem with your fetch operation: ' + error.message);
      dispatch(taskRequestFailed(error))
    });
  };
}