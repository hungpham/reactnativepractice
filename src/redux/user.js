//=============================//
//      Action Types
//=============================//
export const SIGN_IN_REQUEST = 'SIGNIN/REQUESTING';
export const SIGN_IN_SUCCESS = 'SIGNIN/SUCCESSED';
export const SIGN_IN_FAILED = 'SIGNIN/FAILED';
//=============================//
export const SIGN_OUT_REQUEST = 'SIGNOUT/REQUESTING';
export const SIGN_OUT_SUCCESS = 'SIGNOUT/SUCCESSED';
export const SIGN_OUT_FAILED = 'SIGNOUT/FAILED';
//=============================//
export const SIGN_UP_REQUEST = 'SIGNUP/REQUESTING';
export const SIGN_UP_SUCCESS = 'SIGNUP/SUCCESSED';
export const SIGN_UP_FAILED = 'SIGNUP/FAILED';

export const USER_LIST_REQUEST = 'USER_LIST/REQUESTING';
export const USER_LIST_SUCCESS = 'USER_LIST/SUCCESSED';
export const USER_LIST_FAILED = 'USER_LIST/FAILED';

//=============================//
//      Action Creators
//=============================//
export function loginRequest() {
	return {
		type: SIGN_IN_REQUEST
	};
}

export function loginRequestSuccess(json) {
	return {
		type: SIGN_IN_SUCCESS,
		payload: json
	};
}

export function loginRequestFailed(error) {
	return {
		type: SIGN_IN_FAILED,
		error: error.message
	};
}

export function logoutRequest() {
	return {
		type: SIGN_OUT_REQUEST
	};
}

export function logoutRequestSuccess(json) {
	return {
		type: SIGN_OUT_SUCCESS,
		payload: json
	};
}

export function logoutRequestFailed(error) {
	return {
		type: SIGN_OUT_FAILED,
		error: error.message
	};
}
//=============================//
export function signUpRequest() {
	return {
		type: SIGN_UP_REQUEST
	};
}

export function signUpRequestSuccess(json) {
	return {
		type: SIGN_UP_SUCCESS,
		payload: json
	};
}

export function signUpRequestFailed(error) {
	return {
		type: SIGN_UP_FAILED,
		error: error.message
	};
}

//=============================//
export function usersRequest() {
	return {
		type: USER_LIST_REQUEST
	};
}

export function usersRequestSuccess(json) {
	return {
		type: USER_LIST_SUCCESS,
		payload: json
	};
}

export function usersRequestFailed(error) {
	return {
		type: USER_LIST_FAILED,
		error: error.message
	};
}

//=============================//
//      Reducer
//=============================//
export const INITIAL_STATE = {
	error: '',
	loading: false
	// ,user: {
	// 	trainee_id: 3,
	// 	full_name: "test",
	// 	email: "test@gmail.com",
	// 	birthday: "1986-01-20T00:00:00.000Z",
	// 	token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkYXRhIjp7InRyYWluZWVfaWQiOjMsImZ1bGxfbmFtZSI6InRlc3QiLCJlbWFpbCI6InRlc3RAZ21haWwuY29tIiwiYmlydGhkYXkiOiIxOTg2LTAxLTIwVDAwOjAwOjAwLjAwMFoifSwiaWF0IjoxNDg5NDE5Mjg4LCJleHAiOjE0ODk0MjI4ODh9.PutN2HHQGJbGkc-pl7HxwrUkiNdJvHOQjqcmrP3_l-Q"
	// }
};

export function userReducer(state = INITIAL_STATE, action) {
	switch (action.type) {
		case SIGN_IN_REQUEST:
		case SIGN_OUT_REQUEST:
		case SIGN_UP_REQUEST:
		case USER_LIST_REQUEST:
			return {
				...state,
				loading: true,
				error: ''
			};
		case SIGN_IN_FAILED:
		case SIGN_OUT_FAILED:
		case SIGN_UP_FAILED:
		case USER_LIST_FAILED:
			return {
				...state,
				loading: false,
				error: action.error
			};
		case SIGN_IN_SUCCESS:
			return {
				...state,
				loading: false,
				error: '',
				user: action.payload
			};

		case SIGN_OUT_SUCCESS:
			return {
				...state,
				loading: false,
				error: '',
				user: action.payload
			};

		case SIGN_UP_SUCCESS:
			return {
				...state,
				loading: false,
				error: ''
			};
		case USER_LIST_SUCCESS:
			return {
				...state,
				loading: false,
				error: '',
				users: action.payload
			};
		default:
			return state;
	}
};


