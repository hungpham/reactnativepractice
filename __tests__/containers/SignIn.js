import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ConnectedSignIn, { SignIn } from 'containers/SignIn';

//Testing: Connected component
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { INITIAL_STATE as intialState } from 'redux/signin';

//ConnectedSignIn: to test its interaction with Redux - wrapper component returned by connect(), not the App component itself
//SignIn: to test just the rendering of the component

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<SignIn />', () => {

	it('renders correctly', () => {
		const tree = renderer.create(
			<SignIn />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

  it('renders with redux', () => {
    const store = mockStore({ signInReducer: intialState });
    const wrapper = shallow(
      <ConnectedSignIn store={store} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
