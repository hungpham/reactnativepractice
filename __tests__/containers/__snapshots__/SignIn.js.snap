// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> renders correctly 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": null,
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.5,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 45,
            "height": 90,
            "justifyContent": "center",
            "width": 90,
          },
          Object {
            "borderRadius": 60,
            "height": 120,
            "width": 120,
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.25,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#CCC",
          "borderBottomWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "paddingLeft": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "marginLeft": 20,
            "padding": 10,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "resizeMode": "contain",
              "width": 20,
            }
          }
        />
      </View>
      <TextInput
        autoCapitalize="none"
        imageIcon={1}
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="UserName"
        placeholderTextColor="#d7d7d7"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "color": "white",
            "flex": 1,
            "paddingHorizontal": 10,
          }
        }
        underlineColorAndroid="rgba(0, 0, 0, 0)"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#CCC",
          "borderBottomWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "paddingLeft": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "marginLeft": 20,
            "padding": 10,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "resizeMode": "contain",
              "width": 20,
            }
          }
        />
      </View>
      <TextInput
        autoCapitalize="none"
        imageIcon={1}
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#d7d7d7"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "color": "white",
            "flex": 1,
            "paddingHorizontal": 10,
          }
        }
        underlineColorAndroid="rgba(0, 0, 0, 0)"
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "color": "#d7d7d7",
            "paddingRight": 15,
            "paddingTop": 10,
            "textAlign": "right",
          }
        }
      >
        Forgot Password?
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 0.25,
        },
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "red",
          "lineHeight": 20,
          "textAlign": "center",
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ff3366",
            "height": 60,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontSize": 18,
            }
          }
        >
          Sign In
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "color": "#D8D8D8",
          }
        }
      >
        Don't have an account?
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "color": "white",
              "marginLeft": 5,
            }
          }
        >
          Sign Up
        </Text>
      </View>
    </View>
  </View>
</Image>
`;

exports[`<SignIn /> renders with redux 1`] = `
<SignIn
  error=""
  loading={false}
  login={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
